sw/airborne/modules/ctrl/ctrl_eff_sched_rot_wing.c

```diff
@@ -0,0 +1,231 @@
+ */
+
+#include "modules/ctrl/ctrl_eff_sched_rot_wing.h"
+
+#include "firmwares/rotorcraft/stabilization/stabilization_indi.h"
+#include "modules/core/abi.h"
+
+
+#ifndef ROT_WING_EFF_SCHED_IXX_BODY
+#error "NO ROT_WING_EFF_SCHED_IXX_BODY defined"
+#endif
+
+#ifndef ROT_WING_EFF_SCHED_IYY_BODY
+#error "NO ROT_WING_EFF_SCHED_IYY_BODY defined"
+#endif
+
+#ifndef ROT_WING_EFF_SCHED_IZZ
+#error "NO ROT_WING_EFF_SCHED_IZZ defined"
+#endif
+
+#ifndef ROT_WING_EFF_SCHED_IXX_WING
+#error "NO ROT_WING_EFF_SCHED_IXX_WING defined"
+#endif
+
+#ifndef ROT_WING_EFF_SCHED_IYY_WING
+#error "NO ROT_WING_EFF_SCHED_IYY_WING defined"
+#endif
+
+#ifndef ROT_WING_EFF_SCHED_M
+#error "NO ROT_WING_EFF_SCHED_M defined"
+#endif
+
+#ifndef ROT_WING_EFF_SCHED_ROLL_ARM
+#error "NO ROT_WING_EFF_SCHED_ROLL_ARM defined"
+#endif
+
+#ifndef ROT_WING_EFF_SCHED_PITCH_ARM
+#error "NO ROT_WING_EFF_SCHED_PITCH_ARM defined"
+#endif
+
+#ifndef ROT_WING_EFF_SCHED_HOVER_DF_DPPRZ
+#error "NO ROT_WING_EFF_SCHED_HOVER_DF_DPPRZ defined"
+#endif
+
+#ifndef ROT_WING_EFF_SCHED_HOVER_ROLL_PITCH_COEF
+#error "NO ROT_WING_EFF_SCHED_HOVER_ROLL_PITCH_COEF defined"
+#endif
+
+struct rot_wing_eff_sched_param_t eff_sched_p = {
+  .Ixx_body                 = ROT_WING_EFF_SCHED_IXX_BODY,
+  .Iyy_body                 = ROT_WING_EFF_SCHED_IYY_BODY,
+  .Izz                      = ROT_WING_EFF_SCHED_IZZ,
+  .Ixx_wing                 = ROT_WING_EFF_SCHED_IXX_WING,
+  .Iyy_wing                 = ROT_WING_EFF_SCHED_IYY_WING,
+  .m                        = ROT_WING_EFF_SCHED_M,
+  .roll_arm                 = ROT_WING_EFF_SCHED_ROLL_ARM,
+  .pitch_arm                = ROT_WING_EFF_SCHED_PITCH_ARM,
+  .hover_dFdpprz            = ROT_WING_EFF_SCHED_HOVER_DF_DPPRZ,
+  .hover_roll_pitch_coef    = ROT_WING_EFF_SCHED_HOVER_ROLL_PITCH_COEF
+};
+
+struct rot_wing_eff_sched_var_t eff_sched_var;
+
+float rotation_angle_setpoint_deg = 0; // Quad mode
+int16_t rotation_cmd = 9600; // Quad mode
+float pprz_angle_step = 9600. / 45.; // CMD per degree
+
+// Telemetry
+#if PERIODIC_TELEMETRY
+#include "modules/datalink/telemetry.h"
+static void send_rotating_wing_state(struct transport_tx *trans, struct link_device *dev)
+{
+  uint8_t state = 0; // Quadrotor
+  float angle = eff_sched_var.wing_rotation_rad / M_PI * 180.f;
+  pprz_msg_send_ROTATING_WING_STATE(trans, dev, AC_ID,
+                          &state,
+                          &state,
+                          &angle,
+                          &rotation_angle_setpoint_deg,
+                          0,
+                          0);
+}
+#endif
+
+
+/** ABI binding wing position data.
+ */
+#ifndef CTRL_EFF_SCHED_ROT_WING_ID
+#define CTRL_EFF_SCHED_ROT_WING_ID ABI_BROADCAST
+#endif
+PRINT_CONFIG_VAR(CTRL_EFF_SCHED_ROT_WING_ID)
+static abi_event wing_position_ev;
+
+static void wing_position_cb(uint8_t sender_id UNUSED, struct act_feedback_t *pos_msg, uint8_t num_act)
+{
+  for (int i=0; i<num_act; i++){
+    if (pos_msg[i].set.position && (pos_msg[i].idx =  7))
```

Comment:
This hardcoded 7 should definitely not be and should be changed to the generated defines from the airframe file.
Now if we add/change the order in the airframe it will imidiately break.

Reply:
Totally agree. There are still several updates following for this module. The channel is now set in #3133